### PR TITLE
TST: Test pivot_table() with categorical data

### LINKED ISSUE #21370: TST: add additional test cases for pivot_table with categorical data
xref https://github.com/pandas-dev/pandas/pull/21252#discussion_r191665246

Add additional test cases for:

Using the same definition of `df` as you used in your test, `columns`/`values` is incorrect:
```python
In [3]: pd.__version__
Out[3]: '0.23.0'

In [4]: df.pivot_table(columns='A', values='B')
Out[4]:
A  NaN  low
B  2.0  3.0
```

Similarly `index`/`columns`/`values` is incorrect:
```python
In [5]: df['AA'] = df['A']

In [6]: df.pivot_table(index='A', columns='AA', values='B')
Out[6]:
AA   NaN  low
A
NaN  2.0  NaN
low  NaN  3.0
```

### PR DESCRIPTION
**Pull request by ryankarlos**
- [x] closes #21370
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by ryankarlos** [comment]
@jorisvandenbossche @jschendel Just had a go at this - let me know if any changes required

File: pandas/tests/reshape/test_pivot.py [hunk 1]
```diff
@@ -2554,3 +2554,38 @@ def test_margin_normalize(self):
             names=["A", "B"],
         )
         tm.assert_frame_equal(result, expected)
+
+    def test_pivot_with_categorical(self):
+        # gh-21370
```
Comment (jreback) [inline pandas/tests/reshape/test_pivot.py]: can you move this near ```def test_categorical_margins_category(self, observed)``` also if you can accept the observed fixture.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/reshape/test_pivot.py (hunk 1 above)
Comment (ryankarlos) [reply pandas/tests/reshape/test_pivot.py]: Done

Comment (topper-123) [reply pandas/tests/reshape/test_pivot.py]: Can you also use the ``ordered_fixture`` instead of using ``ordered=True``.

Comment (ryankarlos) [reply pandas/tests/reshape/test_pivot.py]: @topper-123 just added and pushed latest changes

**Comment by ryankarlos** [comment]
Not sure why CI is failing

File: pandas/tests/reshape/test_pivot.py [hunk 2]
```diff
@@ -1656,6 +1656,49 @@ def test_categorical_margins_category(self, observed):
         table = df.pivot_table("x", "y", "z", dropna=observed, margins=True)
         tm.assert_frame_equal(table, expected)
 
+    def test_pivot_with_categorical(self, observed, ordered_fixture):
+        # gh-21370
+        idx = [np.nan, "low", "high", "low", np.nan]
```
Comment (jreback) [inline pandas/tests/reshape/test_pivot.py]: you are not using observed here?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/reshape/test_pivot.py (hunk 2 above)
Comment (ryankarlos) [reply pandas/tests/reshape/test_pivot.py]: @jreback Do you mean pass as keyword arg to `pivot_table()` ? Ive just pushed changes

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ryankarlos

**Pull request merged**